lib/rubocop/cop/cop.rb

```diff
@@ -183,8 +189,19 @@ def relevant_file?(file)
           !file_name_matches_any?(file, 'Exclude', false)
       end
 
+      def style_guide_url
+        url = cop_config && cop_config['StyleGuide']
+        (url.nil? || url.empty?) ? nil : url
+      end
+
       private
 
+      def annotate_message(message)
+        message = "#{name}: #{message}" if display_cop_names?
+        message = "#{message} (#{style_guide_url})" if display_style_guide?
+        message
+      end
```

Comment:
I think the word I was looking for was “shadowing” – that the returned `message` can be either the original method argument (if both conditionals are false) or a local variable of the same name.

It’s not that much of a problem, and as @bbatsov pointed out, my code example won’t work.